### ISSUE TITLE
consistent use of severity values for rules and threshold

### ISSUE THREAD

**Issue by UliSei** [MarketSquare/robotframework-robocop#624]
Tested with Robocop 2.1.0:

## Configuring rules
severity values **I/W/E** and **i/w/e** are accepted, e.g.
`-c line-too-long:severity:e`

## Configuring threshold:
only severity values **I/W/E** are accepted.
The setting of e.g.
`--threshold e`
is just ignored.

I would suggest that for severity and threshold the same data set is accepted.

**Comment by bhirsz** [comment]
I agree - it should be consistent (and invalid values for the threshold should be reported). It actually uses the same method of parsing but for some reason we're filtering the values before:

```
class SetRuleThreshold(argparse.Action):
    def __call__(self, parser, namespace, values, option_string=None):
        setattr(namespace, self.dest, RuleSeverity.parser(values) if values in ("I", "W", "E") else RuleSeverity.INFO)
```
It takes only I, W, E values - but ``RuleSeverity/parser`` accepts it in every form (and should raise and error too):
```
    @classmethod
    def parser(cls, value: Union[str, "RuleSeverity"]) -> "RuleSeverity":
        # parser can be invoked from Rule() with severity=RuleSeverity.WARNING (enum directly) or
        # from configuration with severity:W (string representation)
        severity = {
            "error": cls.ERROR,
            "e": cls.ERROR,
            "warning": cls.WARNING,
            "w": cls.WARNING,
            "info": cls.INFO,
            "i": cls.INFO,
        }.get(str(value).lower(), None)
        if severity is None:
            raise ValueError(f"Chose one of: {', '.join(sev.value for sev in cls)}") from None
        return severity
```
It shouldn't be hard to update but I need to take a look if there is any reason it was implemented like this (I could forget specific edge case, but it's most likely just leftover after recent refactors for severity).

**Comment by UliSei** [comment]
Great! I like the solution with reporting when invalid severity values were used.
